modules/data.remote/R/GEDI_AGB_prep.R

```diff
@@ -56,481 +92,437 @@ GEDI_AGB_prep <- function(site_info, time_points, outdir = file.path(getwd(), "G
   AGB_Output <- matrix(NA, length(site_info$site_id), 2*length(time_points)+1) %>% 
     `colnames<-`(c("site_id", paste0(time_points, "_AGB"), paste0(time_points, "_SD"))) %>% as.data.frame()#we need: site_id, AGB, std, target time point.
   AGB_Output$site_id <- site_info$site_id
-  # loop over each time point
+  # loop over each time point.
   for (i in seq_along(time_points)) {
     # create start and end dates.
     start_date <- seq(time_points[i], length.out = 2, by = paste0("-", search_window))[2]
     end_date <- seq(time_points[i], length.out = 2, by = search_window)[2]
-    # extract GEDI AGB.
-    AGB <- GEDI_AGB_extract(site_info, start_date, end_date, outdir, nfile.min = 0, nrow.min = 1, buffer = as.numeric(buffer))
-    for (j in seq_along(AGB)) {
+    # download GEDI tiles.
+    files <- NASA_DAAC_download(ul_lat = bbox[4], 
+                                ul_lon = bbox[1], 
+                                lr_lat = bbox[3], 
+                                lr_lon = bbox[2], 
+                                ncore = parallel::detectCores(), 
```

Comment:
Shouldn't this be passed in from the outer function?
```suggestion
                                ncore = cores, 
```